processing/dwf_calculation_algorithm.py

```diff
@@ -0,0 +1,396 @@
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+"""
+
+__author__ = 'Nelen en Schuurmans'
+__date__ = '2021-01-27'
+__copyright__ = '(C) 2021 by Nelen en Schuurmans'
+
+# This will get replaced with a git SHA1 when you do a git archive
+
+__revision__ = '$Format:%H$'
+
+
+import os
+import sqlite3
+import json
+import csv
+from datetime import datetime
+
+from qgis.PyQt.QtCore import QCoreApplication
+from qgis.core import (QgsProcessing,
+                       QgsFeatureSink,
+                       QgsProcessingAlgorithm,
+                       QgsProcessingParameterFeatureSource,
+                       QgsProcessingParameterFeatureSink,
+                       QgsProcessingParameterFile,
+                       QgsProcessingParameterString,
+                       QgsProcessingParameterFileDestination,
+                       QgsProcessingParameterProviderConnection,
+                       QgsProviderRegistry,
+                       QgsProviderConnectionException,
+                       QgsProcessingException)
+
+
+
+# OUTPUT : lateral as a json file 
+
+# {
+#   "offset": 180,
+#   "interpolate": false,
+#   "values": [
+#     [0.0, 0.0002],
+#     [900.0, 0.0],
+#     [1200.0, 0.0001],
+#     [1620.0, 0.0]
+#   ],
+#   "units": "m3/s",
+#   "connection_node": 4
+# }
+
+# INPUT CSV FOR API CLIENT :
+# FORMAT : lat_id, connection_node_id, timeseries
+
+# Default values
+DWF_FACTORS = [
+    [0, 0.03],
+    [1, 0.015],
+    [2, 0.01],
+    [3, 0.01],
+    [4, 0.005],
+    [5, 0.005],
+    [6, 0.025],
+    [7, 0.080],
+    [8, 0.075],
+    [9, 0.06],
+    [10, 0.055],
+    [11, 0.05],
+    [12, 0.045],
+    [13, 0.04],
+    [14, 0.04],
+    [15, 0.035],
+    [16, 0.035],
+    [17, 0.04],
+    [18, 0.055],
+    [19, 0.08],
+    [20, 0.07],
+    [21, 0.055],
+    [22, 0.045],
+    [23, 0.04],
+]
+
+
+def get_dwf_factors_from_file(file_path):
+    
+    dwf_factors = []
+    
+    with open(file_path) as csv_file:
+        reader = csv.reader(csv_file, delimiter = ',')
+        for row in reader:
+            print(row)
+            dwf_factors += [[int(row[0]), float(row[1])]]
+
+    return dwf_factors
+    
+
+
+def read_dwf_per_node(spatialite_path):
+    
+    """Obtains the 24h dry weather flow per connection node from the a 3Di model sqlite-file."""
+    conn = sqlite3.connect(spatialite_path)
+    c = conn.cursor()
+
+    # DWF per person = 120 l/inhabitant / 1000 = 0.12 m3/inhabitant
+    dwfPerPerson = 0.12
```

Comment:
This is best done with a constant at the start of the file.

And.... don't mix theMixedCaseStyle with python's preferred `under_score_style`, so make it `DWF_PER_PERSON`.